sklearn/model_selection/tests/test_validation.py

```diff
@@ -740,3 +741,70 @@ def test_cross_val_predict_sparse_prediction():
     preds_sparse = cross_val_predict(classif, X_sparse, y_sparse, cv=10)
     preds_sparse = preds_sparse.toarray()
     assert_array_almost_equal(preds_sparse, preds)
+
+
+def test_cross_val_predict_apply_func_cv():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    classes = len(set(y))
+
+    # Shuffle the data before doing cross validation
+    shf = np.random.permutation(np.arange(len(X)))
+    X = X[shf]
+    y = y[shf]
+
+    kfold = KFold(len(iris.target))
+
+    for func_name in APPLY_FUNC_NAMES:
+        cv_generator = kfold.split(X, y)
+
+        est = LogisticRegression()
+
+        predictions = cross_val_predict(est, X, y, apply_func=func_name)
+        assert_equal(len(predictions), len(y))
+
+        expected_predictions = np.zeros([len(y)]) if func_name == 'predict' else np.zeros([len(y), classes])
+        func = getattr(est, func_name)
+
+        # Naive loop (should be same as cross_val_predict):
+        for train, test in cv_generator:
+            est.fit(X[train], y[train])
+            expected_predictions[test] = func(X[test])
+
+        predictions = cross_val_predict(est, X, y, apply_func=func_name, cv=kfold)
+        assert_array_almost_equal(expected_predictions, predictions, err_msg='when func_name=%s, expected size=%s,'
+                                                                             'size=%s'
+                                  % (func_name, expected_predictions.size, predictions.size))
+
+
+def test_cross_val_predict_apply_func_leave_one_out():
```

Comment:
I wouldn't have bothered testing this case given that it's already tested for the predict case. You can't test every possible combination of inputs.